homeassistant/components/becker/cover.py

```diff
@@ -0,0 +1,191 @@
+"""Support for Becker RF covers."""
+
+import logging
+
+import voluptuous as vol
+
+from homeassistant.components.cover import (
+    SUPPORT_CLOSE,
+    SUPPORT_CLOSE_TILT,
+    SUPPORT_OPEN,
+    SUPPORT_OPEN_TILT,
+    SUPPORT_STOP,
+    CoverEntity,
+)
+from homeassistant.const import (
+    CONF_COVERS,
+    CONF_NAME,
+    CONF_VALUE_TEMPLATE,
+    STATE_CLOSED,
+    STATE_OPEN,
+)
+from homeassistant.exceptions import TemplateError
+import homeassistant.helpers.config_validation as cv
+from homeassistant.helpers.restore_state import RestoreEntity
+
+from .const import (
+    CLOSED_POSITION,
+    CONF_CHANNEL,
+    DEVICE_CLASS,
+    DOMAIN,
+    INTERMEDIATE_POSITION,
+    OPEN_POSITION,
+    VENTILATION_POSITION,
+)
+from .utils import extract_entities, initialise_templates
+
+_LOGGER = logging.getLogger(__name__)
+
+COVER_FEATURES = (
+    SUPPORT_OPEN | SUPPORT_CLOSE | SUPPORT_STOP | SUPPORT_OPEN_TILT | SUPPORT_CLOSE_TILT
+)
+
+
+_VALID_STATES = [STATE_OPEN, STATE_CLOSED, "true", "false"]
+
+COVER_SCHEMA = vol.Schema(
+    {
+        vol.Optional(CONF_NAME): cv.string,
+        vol.Required(CONF_CHANNEL): cv.string,
+        vol.Optional(CONF_VALUE_TEMPLATE): cv.template,
+    }
+)
+
+
+async def async_setup_entry(hass, config_entry, async_add_entities):
+    """Load Becker Covers based on a config entry."""
+    covers = []
+
+    becker_config = hass.data[DOMAIN]
+    becker_connector = becker_config["connector"]
+
+    for channel, device_config in becker_config[CONF_COVERS].items():
+        name = device_config[CONF_NAME]
+        _LOGGER.debug("SETUP cover on channel {channel} with name {name}")
+
+        state_template = device_config.get(CONF_VALUE_TEMPLATE)
+        if channel is None:
+            _LOGGER.error("Must specify {CONF_CHANNEL}")
+            continue
+        templates = {
+            CONF_VALUE_TEMPLATE: state_template,
```

Comment:
Can you explain what is going on here? 

A user should not be required to implement their own templates.

Reply:
This is used if you want to template the state of a cover (for instance with a open/close sensor)

Reply:
That's not something we should support inside the integration that represents the data from the API. Such features need to be added in a template sensor that forwards commands to this entity.

Reply:
Ok I've remove everything about templates for cover positions then